### PR TITLE
WIP: Render osImageURL, handle "bootstrap" case in MCD 

### PR DESCRIPTION
**Pull request by cgwalters**
This is option 2 for https://github.com/openshift/machine-config-operator/pull/273

Still a WIP, it builds but haven't tested updates with it.

### CODE REVIEW

File: pkg/controller/template/render.go [hunk 1]
```diff
@@ -91,21 +91,16 @@ func generateMachineConfigs(config *RenderConfig, templateDir string) ([]*mcfgv1
 	return cfgs, nil
 }
 
+// GenerateMachineConfigsForRole is part of generateMachineConfigs; it operates
```
Comment (ashcrow) [inline pkg/controller/template/render.go]: Thank you for adding the documentation!

**Comment by ashcrow** [comment]
Terraform/AWS flake **and**
```
level=fatal msg="waiting for openshift-console URL: context deadline exceeded"
```

/retest

**Comment by cgwalters** [comment]
OK rolled in https://github.com/openshift/machine-config-operator/pull/329 to this and also added a hacky patch to give us some debugging data when our e2e fails.

**Comment by ashcrow** [comment]
ISE trying to see logs

/retest

**Comment by ashcrow** [comment]
```
http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug=""
error: watch closed before Until timeout
```

/retest

**Comment by cgwalters** [comment]
OK so here's the problem I'm struggling with...the "bootstrap MC" and the "inital rendered MC" need to be the same.  If we land [the installer change](https://github.com/openshift/installer/pull/1149) without this PR, or vice versa - things will break because booted nodes won't be able to find the MC they expect.

That was the core problem that brought this PR to a halt.

We need to figure out how to "ratchet" the PRs to both the installer and here.

What I'm vaguely thinking about is that we somehow detect whether or not the installer set up the osimageurl.  Maybe stuff something in the `install-config-v1` configmap?  If it didn't, then we also don't render in the non-bootstrap path?

**Comment by cgwalters** [comment]
Splitting this PR into:

https://github.com/openshift/machine-config-operator/pull/362
https://github.com/openshift/machine-config-operator/pull/363

**Pull request closed without merging**